pkg/client/client.go

```diff
@@ -178,10 +180,18 @@ func NewClientFromKeys(accessKey, secretAccessKey, region string) (Client, error
 	if err != nil {
 		return nil, err
 	}
+	s.Handlers.Build.PushBackNamed(addProviderVersionToUserAgent)
 
 	return &awsClient{
 		ec2Client:   ec2.New(s),
 		elbClient:   elb.New(s),
 		elbv2Client: elbv2.New(s),
 	}, nil
 }
+
+// addProviderVersionToUserAgent is a named handler that will add cluster-api-provider-aws
+// version information to requests made by the AWS SDK.
+var addProviderVersionToUserAgent = request.NamedHandler{
+	Name: "openshift.io/cluster-api-provider-aws",
+	Fn:   request.MakeAddToUserAgentHandler("openshift.io cluster-api-provider-aws", "dummy"),
```

Comment:
Maybe I dumb question but why does the name have '/' but Fn not?

Reply:
None of that is a blocker for me. Though, we might even change the `dummy` literal into more customer legit string.

Reply:
> Maybe a dumb question but why does the name have '/' but Fn not?

https://docs.aws.amazon.com/sdk-for-go/api/aws/request/#MakeAddToUserAgentHandler
the useragent is generated based on Fn arguments:
```
name/version (extra0; extra1; ...)
```
therefore the arg to Fn doesn't use `/` in the name.

As for the Handler's Name, i didn't want to put spaces.

> None of that is a blocker for me. Though, we might even change the `dummy` literal into more customer legit string.

The cluster-api-provider-aws has no usable variable for version. That's why it is dummy.

Reply:
We can replace the `dummy` with version once we implemented the `version` subcommand.

@abhinavdahiya thanks for the rationale